subsys/bluetooth/audio/audio_iso.c

```diff
@@ -0,0 +1,177 @@
+/* Common functions for LE Audio services */
+
+/*
+ * Copyright (c) 2022 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "audio_iso.h"
+
+static struct bt_audio_iso iso_pool[CONFIG_BT_ISO_MAX_CHAN];
+static struct bt_audio_iso_unicast unicast_pool[CONFIG_BT_ISO_MAX_CHAN];
+
+static uint8_t bt_audio_iso_index(struct bt_audio_iso *audio_iso)
+{
+	ptrdiff_t index = 0;
+
+	index = audio_iso - iso_pool;
+	__ASSERT(index >= 0 && index < ARRAY_SIZE(iso_pool),
+		 "Invalid bt_audio_iso pointer");
+
+	return (uint8_t)index;
+}
+
+static struct bt_audio_iso *bt_audio_iso_new(void)
+{
+	struct bt_audio_iso *audio_iso = NULL;
+	int i;
+
+	for (i = 0; i < ARRAY_SIZE(iso_pool); i++) {
+		if (atomic_cas(&iso_pool[i].ref, 0, 1)) {
+			audio_iso = &iso_pool[i];
+			break;
+		}
+	}
+
+	if (!audio_iso) {
+		return NULL;
+	}
+
+	(void)memset(audio_iso, 0, offsetof(struct bt_audio_iso, ref));
+
+	return audio_iso;
+}
+
+struct bt_audio_iso *bt_audio_iso_broadcast_new(void)
+{
+	struct bt_audio_iso *audio_iso;
+
+	audio_iso = bt_audio_iso_new();
+	if (audio_iso == NULL) {
+		return NULL;
+	}
+
+	audio_iso->type = BT_AUDIO_ISO_TYPE_BROADCAST;
+
+	return audio_iso;
+}
+
+struct bt_audio_iso *bt_audio_iso_unicast_new(void)
+{
+	struct bt_audio_iso_unicast *unicast;
+	struct bt_audio_iso *audio_iso;
+
+	audio_iso = bt_audio_iso_new();
+	if (audio_iso == NULL) {
+		return NULL;
+	}
+
+	unicast = &unicast_pool[bt_audio_iso_index(audio_iso)];
+	__ASSERT_NO_MSG(unicast->acl == NULL);
```

Comment:
I'm just wondering what the rational is for using `__ASSERT_NO_MSG` instead of `__ASSERT` ?

Reply:
As far as I remember, assert prints the asserted line anyway. The check here is pretty straightforward, so I assumed additional message is not needed. Anyway, I changed the code so that there's no assert here anymore